services/rbac/service.go

```diff
@@ -0,0 +1,176 @@
+package rbac
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
```

Comment:
I don't know how up to date is the https://jwt.io/. Maybe this library already implemented those checks. But would nice to check.
Also, we are checking token agains database/provider's, right?

Reply:
There's 2 things, we go through an oauth grant for gaining access to a users details (from Google, Microsoft, etc) and then trust that this person is who he/she says she is (effectively externalizing the identity trust) and from there on generate our own JWT to secure communications with the Estafette API.

What the jwt-go library is lacking is at least an option to set clock skew. And we can set a couple of extra claims and check those in an incoming token, for example the `issued at time` and the `issuer`: https://github.com/estafette/estafette-ci-api/blob/master/auth/helpers.go#L180-L182

Currently there's a check to see if the token is expired, but that's all: https://github.com/appleboy/gin-jwt/blob/master/auth_jwt.go#L373-L386